charts/datadog/templates/_daemonset-volumes-linux.yaml

```
@@ -51,6 +51,11 @@
     path: /usr/src
   name: src
 {{- end }}
+{{- if .Values.datadog.systemProbe.enableRuntimeCompiler }}
+- hostPath:
+    path: {{ .Values.datadog.systemProbe.runtimeCompilerOutputDir }}
+  name: runtime-compiler-output-dir
```

it is better to add also the `type: DirectoryOrCreate` if the folder doesn't exist yet

```suggestion
- hostPath:
    path: {{ .Values.datadog.systemProbe.runtimeCompilerOutputDir }}
    type: DirectoryOrCreate
  name: runtime-compiler-output-dir
```